### ISSUE TITLE
CR not handling closed ports - Debian

### ISSUE THREAD

**Issue by Azaret** [CentralReport/CentralReport#65]
Hi,

When some ports are not open, Central Report run without warning the user. Also the exception fired by python when a port is closed seems to be send only on debug logs, and not on error logs.

```
[04/10/2013 09:42:41] INFO:      ------------------------------------------------
[04/10/2013 09:42:41] INFO:      CentralReport is starting...
[04/10/2013 09:42:41] INFO:      Current user: azaret
[04/10/2013 09:42:41] DEBUG:     Linux config
[04/10/2013 09:42:41] DEBUG:     Configuration file: Found. Reading it.
[04/10/2013 09:42:41] DEBUG:     Reading the config file...
[04/10/2013 09:42:41] INFO:      Debian detected. Starting ThreadChecks...
[04/10/2013 09:42:41] DEBUG:     ThreadChecks is starting...
[04/10/2013 09:42:41] DEBUG:     Interval between two checks: 60 seconds
[04/10/2013 09:42:42] INFO:      Enabling the webserver...
[04/10/2013 09:42:42] INFO:      CentralReport started!
[04/10/2013 09:42:42] DEBUG:     ---- New check -----
[04/10/2013 09:42:42] DEBUG:     Doing a CPU check...
[04/10/2013 09:42:43] DEBUG:     Doing a memory check...
[04/10/2013 09:42:43] DEBUG:     Doing a load average check...
[04/10/2013 09:42:43] DEBUG:     Doing a disk check...
[04/10/2013 09:42:43] DEBUG:     All checks are done
[04/10/2013 09:42:43] DEBUG:     Next checks in 60 seconds...
Exception in thread Thread-2:
Traceback (most recent call last):
  File "/usr/lib/python2.6/threading.py", line 532, in __bootstrap_inner
    self.run()
  File "/usr/local/lib/centralreport/web/server.py", line 94, in run
    cherrypy.server.start()
  File "/usr/local/lib/centralreport/libs/cherrypy-3.2.2.zip/cherrypy/_cpserver.py", line 151, in start
    ServerAdapter.start(self)
  File "/usr/local/lib/centralreport/libs/cherrypy-3.2.2.zip/cherrypy/process/servers.py", line 167, in start
    wait_for_free_port(*self.bind_addr)
  File "/usr/local/lib/centralreport/libs/cherrypy-3.2.2.zip/cherrypy/process/servers.py", line 410, in wait_for_free_port
    raise IOError("Port %r not free on %r" % (port, host))
IOError: Port 8080 not free on '0.0.0.0'
```

**Comment by miniche** [comment]
Yes, this point must be improved.
I've added this on our todo list. It will be available in a next release (perhaps in the 0.2.0, which coming in the next weeks).

Stay tuned :)

**Comment by miniche** [comment]
Planned for the version 0.3.0. The test will be added before starting the web part.

**Comment by miniche** [comment]
Now, when the chosen port is already in use on the current host, the web server is not started anymore.
The following message is logged in /var/log/centralreport/:

```
[04/12/2013 06:38:01] ERROR:     Error launching the webserver: port 8080 is already in use on this host!
```

Available in the develop branch, and in the future version 0.3.0.

**Comment by miniche** [comment]
Solved by #72.

Comments are always welcome about this issue, or please open a new issue for another case :+1:
